src/openms/include/OpenMS/CHEMISTRY/MASSDECOMPOSITION/IMS/RealMassDecomposer.h

```diff
@@ -126,7 +126,7 @@ namespace OpenMS
         Decomposer to be used for exact decomposing using
         integer arithmetic.
       */
-      std::auto_ptr<integer_decomposer_type> decomposer_;
+      std::shared_ptr<integer_decomposer_type> decomposer_;
```

Comment:
Ok I see. Fine with me. I was just asking because I saw a video on C11 smart pointers (https://www.youtube.com/watch?v=JfmTagWcqoE). As I understood it, it is better to use unique_ptrs in this case and perform deep copies.